app/classes/Transvision/Search.php

```diff
@@ -267,4 +310,92 @@ public function getRepository()
     {
         return $this->repository;
     }
+
+    /**
+     * Set the type of search we want to perform
+     *
+     * @param  string $type The type of search we will perform
+     * @return $this
+     */
+    public function setSearchType($type)
+    {
+        if (in_array($type, $this->search_types)) {
+            $this->search_type = $type;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get the type of search we want to perform
+     *
+     * @return string Type of search
+     */
+    public function getSearchType()
+    {
+        return $this->search_type;
+    }
+
+    /**
+     * Get all the types of search we can perform
+     *
+     * @return array Types of search
+     */
+    public function getSearchTypes()
+    {
+        return $this->search_types;
+    }
+
+    /**
+     * Get all the options we use in the search form
+     *
+     * @return array Form options
+     */
+    public function getFormSearchOptions()
+    {
+        return $this->form_search_options;
+    }
+
+    /**
+     * Get all the checkboxes we use in the search form
+     *
+     * @return array Form checkboxes
+     */
+    public function getFormCheckboxes()
+    {
+        return $this->form_checkboxes;
+    }
+
+    /**
+     * Set the Locales we will use for the search
+     *
+     * @param  array $locales The locale codes
+     * @return $this
+     */
+    public function setLocales($locales)
+    {
+        // We only allow up to 3 locales for analysis
+        $locales = array_slice($locales, 0, 3);
+
+        if (count($locales) == 3) {
+            // We check if the 3rd locale is the same as the 2nd one
+            if ($locales[2] == $locales[1]) {
+                unset($locales[2]);
+            }
+        }
+
+        $this->locales = $locales;
+
+        return $this;
+    }
+
+    /**
+     * Get the locales used for searching
+     *
+     * @return array Locale codes
+     */
+    public function getLocales()
```

Comment:
In the code the function is always used as `$search->getLocales()[0]$search->getLocales()[NUMBER]`, which is a bit confusing. I think it would be clearer to have calls like `$search->getLocales()[0]$search->getLocales('source')` to get the specific locale: `source`, ´target´, `extra`, `all` in case we need the entire array (I don't think we do).